Minor front-end refactoring

absentabyss opened the pull request:
Front-end now processes requests and updates the content instead of redirecting the user to raw JSON pages.

absentabyss committed:
[FRONT] Refactored index.html scripts into scripts and fixed basic functionality

All front-end elements work as originally intended.

absentabyss committed:
[BACK] Bug fixes and minor refactoring

absentabyss committed:
IMPORTANT: MAJOR CHANGE!

Changed tabs for spaces.

absentabyss committed:
[Front] Minor formatting correction

Replaced tabs with spaces and added the "var" keyword.

absentabyss committed:
[Front] Server properties are properly submitted

Submission button no longer redirects to JSON file.
Added a timestamp as feedback for successful submission.

absentabyss committed:
Added favicon

absentabyss committed:
[Back] Fixed typo in variable name

absentabyss committed:
[Front] Removed actions from HTML forms

These actions would take users to JSON pages if JavaScript didn't load
properly.

absentabyss committed:
[Front] setServerProperties now returns message on failure

absentabyss committed:
[Front] Removed debugging console.logs

absentabyss committed:
[Front] Added function that creates HTML for buttons

absentabyss committed:
[Back] deleteWorld.py now deletes .zips

absentabyss committed:
[Front] Level management module now handles JSON responses

absentabyss committed:
Fixed Bug in deleteLevel function

The function would fail if it had had no requests for download because
it would try to delete a missing zip.

absentabyss committed:
[Front] Fixed bug in deleteWorld

When setting multiple levels a listener would be added to the delete and
download functions without clearing the old listeners. This would cause
the delete and download buttons to send multiple requests per click.

absentabyss committed:
[Front] The level list now updates after server is stopped

absentabyss committed:
[Front] Minor change to design and unified scripts

Moved input labels to the other side of radio buttons.
getVersion.js and versionLinks.js were merged into versionManagement.js

absentabyss committed:
[Front] Version management doesn't redirect to JSON pages

No version features redirect to JSON pages, they now process back-end
responses instead. Also, getVersion.js and versionLinks.js were unified
into versionManagement.js

absentabyss committed:
Minor error handling in serverProperties.js

getServerProperties.py's response could be missing a "server-properties"
attribute. In such cases, the function exits gracefully.

absentabyss committed:
[Testing] Updated testing for new front-end

absentabyss committed:
Removed unnecessary console logging

pidement commented on cgi-bin/sendConsoleCommand.py:
```diff
@@ -8,7 +8,7 @@
 def core(response):
     if utils.server_is_alive():
-        command = base64.b64decode(commandEncoded).decode('utf-8')
+        command = base64.b64decode(command_encoded).decode('utf-8')
```
why the change of this variable only here?

absentabyss replied:
Snake case is consistently used within Python scripts.

pidement replied:
Yes I know, but why the rename It's only on this variable? before this PR the variable it's not used?

absentabyss replied:
It's probably a fix from a previous refactoring. When I moved a lot of functions into utils.py I think I introduced a Bug by creating a command_encoded variable and forgot to rename this one instance.

pidement commented on index.html:
```diff
@@ -1,400 +1,287 @@
+            <div>
+                <input type="submit" value="Save" /><label
+                    id="server-properties-success"
+                ></label>
```
This indentation it's a little confusing.

absentabyss replied:
Indentation is handled by Prettier. We can override it if you want.

pidement replied:
I think that
```html
<input type="submit" value="Save" />
<label id="server-properties-sucess"></label>
```
it's better than:
```html
<input type="submit" value="Save" /><label
id="server-properties-sucess"
></label>
```
Can we make this change?

absentabyss replied:
Yes. What should we do with:
```html
></script>
```
? Personally, I don't mind it.

absentabyss replied:
Script opening and closing tags are now in the same line.

pidement commented on index.html:
```diff
@@ -1,400 +1,287 @@
+            <br />
+            <input name="submit" type="submit" value="Upload Level" /><label
+                id="upload-level-success"
+            ></label>
```
same. I guess that it's the default vim indentation. Can we change it?

absentabyss replied:
Refer to my first reply on the issue.

pidement replied:
I'll put this as resolved.

pidement commented on index.html:
```diff
@@ -1,400 +1,287 @@
+            type="text/javascript"
+            src="./scripts/serverProperties.js"
+        ></script>
+        <script type="text/javascript" src="./scripts/publicIp.js"></script>
```
The same here.
```></script>```

absentabyss replied:
Refer to my first reply on the issue.

pidement commented on scripts/utils.js:
```diff
@@ -0,0 +1,60 @@
+function refreshElement(element) {
+    var new_element = element.cloneNode(true);
+    element.parentNode.replaceChild(new_element, element);
+}
```
why this function it's needed?

absentabyss replied:
I think it's explained in the Commit. Elements can have listeners, this is a way to clear listeners that have been previously added to the element (and more). There's another method that removes specific listeners, but some listeners have a Lambda callback so I couldn't use that method.

pidement commented on scripts/levelManagement.js:
```diff
@@ -0,0 +1,136 @@
+    refreshElement(downloadLevelButton);
+    var downloadLevelButton = document.getElementById(
+        "selected-level-download"
+    );
```
this element it's defined twice.

absentabyss replied:
That's because I wasn't sure if refreshElement destroyed the old element or not. I've had problems with similar functions in the past, so I did that just to be safe.

pidement commented on scripts/levelManagement.js:
```diff
@@ -0,0 +1,136 @@
+            }
+        };
+        request.onerror = function () {};
+        request.send(); // create FormData from form that triggered event
```
why not use ```httpGetAsync``` here ?

absentabyss replied:
Whoops!

pidement commented on scripts/levelManagement.js:
```diff
@@ -0,0 +1,136 @@
+            }
+        };
+        request.onerror = function () {};
+        request.send(); // create FormData from form that triggered event
```
Why do not use ```httpGetAsync``` here ?

absentabyss replied:
Whoops!

pidement commented on scripts/levelManagement.js:
```diff
@@ -0,0 +1,136 @@
+    });
+}
+
+function getSelectedLevel(responseJSON) {
```
I think that we can rename this function to something like ```prepareSelectedLevel``` or ```setSelectedLever```

absentabyss replied:
Maybe `refreshSelectedLevelModule`?

pidement commented on scripts/levelManagement.js:
```diff
@@ -0,0 +1,136 @@
+    configureDeleteLevelButton();
+}
+
+function getAvailableLevels(responseJSON) {
```
The same here. This function modifies some elements and don't return a value.

absentabyss replied:
Could you specify the issue? I don't see a problem with not returning anything under the current implementation.

pidement replied:
This functions edit some elements of the UI. So I think that it is better a name like ```updateAvailableLevels``` or something like that.

pidement commented on scripts/levelManagement.js:
```diff
@@ -0,0 +1,136 @@
+
+function updateLevelModule() {
+    httpGetAsync("/cgi-bin/getCurrentLevel.py", getSelectedLevel);
+    httpGetAsync("/cgi-bin/getLevelsList.py", getAvailableLevels);
```
Another possible name for the functions mentioned above are ```updateSelectedLevel``` and ```updateAvailableLevels```

absentabyss replied:
I agree.

pidement commented on scripts/utils.js:
```diff
@@ -0,0 +1,60 @@
+ * }
+ * // It can be attached to a form with the following command.
+ * document.getElementById(formId).addEventListener("submit", formSubmit);
+ */
```
remove commented code

absentabyss replied:
It's a preset for sending forms through JavaScript. The URL is different for each form, and I don't know how to pass URL as an argument on event listeners.

pidement commented on scripts/versionManagement.js:
```diff
@@ -0,0 +1,65 @@
+                }
+            };
+            request.onerror = function () {};
+            request.send(); // create FormData from form that triggered event
```
Why not use ```httpGetAsync``` ?

absentabyss replied:
Whoops!

absentabyss committed:
Replaced cluttered code with httpGetAsync

absentabyss committed:
Commented index.html

absentabyss committed:
[Code style] Minor improvement

absentabyss committed:
Minor improvement in coding style

absentabyss committed:
Renamed folder

absentabyss committed:
Documenting and minor improvement in get_server_properties

absentabyss committed:
Added EULA disclaimer

absentabyss committed:
Renamed functions to promote "get" convention

Get convention: the "get" word is reserved for fetching data. A function
that has "get" in its name is expected not to make any changes to the
back or front-end.

absentabyss committed:
Fixed bug with getCurrentDatetime

The function was too verbose and it rendered times like 09:05 as 9:5.

pidement reviewed (approved):
👏 Nice Work!
